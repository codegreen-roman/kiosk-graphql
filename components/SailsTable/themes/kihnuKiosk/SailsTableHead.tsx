import React, { FC } from 'react'
import { Icon } from '@components/Icon'
import { IconHolder } from './SailsTableHeadStyles'
import { TicketsAmount } from '@components/SailsTable'
import { Box, Grid } from '@material-ui/core'

export interface SailsTableHeadProps {
  maxTickets: TicketsAmount
}

const SailsTableHead: FC = () => (
  <Grid item xs={12}>
    <Box padding="7px 25px">
      <Grid container spacing={0} justify="space-between" alignItems="center">
        <Grid item xs={4}>
          <Box paddingLeft="10px">
            <IconHolder as={Icon.clock} gray height="24px" />
          </Box>
        </Grid>
        <Grid item xs={8}>
          <Grid container spacing={0} justify="space-between" alignItems="center">
            <Grid item xs={4}>
              <IconHolder as={Icon.passenger} height="20px" />
            </Grid>
            <Grid item xs={4}>
              <IconHolder as={Icon.vehicle} height="15px" />
            </Grid>
            <Grid item xs={4}>
              <IconHolder as={Icon.bicycle} height="15px" />
            </Grid>
          </Grid>
        </Grid>
      </Grid>
    </Box>
  </Grid>
)

export default SailsTableHead
