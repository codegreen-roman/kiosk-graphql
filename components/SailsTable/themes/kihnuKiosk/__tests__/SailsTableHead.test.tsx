import React, { ReactNode } from 'react'
import SailsTableHead from '../SailsTableHead'
import { shallow } from 'enzyme'

describe('SailsTableHead', () => {
  function render(): ReactNode {
    return shallow(<SailsTableHead />)
  }

  describe('when default props passed', () => {
    it('matches snapshot', () => {
      expect(render()).toMatchSnapshot()
    })
  })
})
