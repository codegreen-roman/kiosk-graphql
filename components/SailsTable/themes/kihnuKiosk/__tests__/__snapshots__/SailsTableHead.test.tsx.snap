// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SailsTableHead when default props passed matches snapshot 1`] = `
<WithStyles(ForwardRef(Grid))
  item={true}
  xs={12}
>
  <Styled(MuiBox)
    padding="7px 25px"
  >
    <WithStyles(ForwardRef(Grid))
      alignItems="center"
      container={true}
      justify="space-between"
      spacing={0}
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={4}
      >
        <Styled(MuiBox)
          paddingLeft="10px"
        >
          <SailsTableHeadStyles__IconHolder
            as={[Function]}
            gray={true}
            height="24px"
          />
        </Styled(MuiBox)>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={8}
      >
        <WithStyles(ForwardRef(Grid))
          alignItems="center"
          container={true}
          justify="space-between"
          spacing={0}
        >
          <WithStyles(ForwardRef(Grid))
            item={true}
            xs={4}
          >
            <SailsTableHeadStyles__IconHolder
              as={[Function]}
              height="20px"
            />
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid))
            item={true}
            xs={4}
          >
            <SailsTableHeadStyles__IconHolder
              as={[Function]}
              height="15px"
            />
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid))
            item={true}
            xs={4}
          >
            <SailsTableHeadStyles__IconHolder
              as={[Function]}
              height="15px"
            />
          </WithStyles(ForwardRef(Grid))>
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </Styled(MuiBox)>
</WithStyles(ForwardRef(Grid))>
`;
